suites/libraries/cms/console/CheckJoomlaUpdatesCommandTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * @package	    Joomla.UnitTest
+ * @subpackage  Version
+ *
+ * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license	    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+/**
+ * Test class for CheckJoomlaUpdatesCommand.
+ *
+ * @package     Joomla.UnitTest
+ * @subpackage  Version
+ * @since       3.0
+ */
+
+use Joomla\CMS\Console\CheckJoomlaUpdatesCommand;
+
+class CheckJoomlaUpdatesCommandTest extends \PHPUnit\Framework\TestCase
+{
+	/**
+	 * Object under test
+	 *
+	 * @var    CheckJoomlaUpdatesCommand
+	 * @since  3.0
+	 */
+	protected $object;
+
+	/**
+	 * Sets up the fixture, for example, opens a network connection.
+	 * This method is called before a test is executed.
+	 *
+	 * @return  void
+	 *
+	 * @since   3.0
+	 */
+	protected function setUp()
+	{
+		$this->object = new CheckJoomlaUpdatesCommand;
+	}
+
+	/**
+	 * Overrides the parent tearDown method.
+	 *
+	 * @return  void
+	 *
+	 * @see     \PHPUnit\Framework\TestCase::tearDown()
+	 * @since   3.6
+	 */
+	protected function tearDown()
+	{
+		unset($this->object);
+		parent::tearDown();
+	}
+
+	/**
+	 * Check if Update Information Contains required information
+	 *
+	 * @since version 4.0
+	 */
+	public function testIfUpdateInformationContainsRequiredParameters()
+	{
+		$updateInfo = $this->object->getUpdateInformation();
+		$this->assertArrayHasKey('installed', $updateInfo, 'The current version is not defined.');
+		$this->assertArrayHasKey('latest', $updateInfo, 'The latest Version must be defined.');
+		$this->assertArrayHasKey('object', $updateInfo, 'Update details information should be set.');
+		$this->assertArrayHasKey('hasUpdate', $updateInfo);
+	}
+
+	/**
+	 * Test if installed version returned from Update Information is JVersion
+	 *
+	 * @since version 4.0
+	 */
+	public function testIfJversionIsInstalledVersion()
+	{
+		$this->assertEquals(\JVERSION, $updateInfo['installed']);
+	}
+
+	/**
+	 * Tests if the name of the command is 'check-updates'
+	 *
+	 * @since version 4.0
+	 */
+	public function testIfCommandNameIsSpecifiedProperly()
+	{
```

Comment:
i don't understand

Reply:
You should test the code you have written and not funtionality of another class. I fact your are testing if within the initialise method the setName part has set the correct value. You could argue that testing if the correct value is set is something we need to check. I would say it doesn't matter, because if doesn't break the functionality of the class under test. It would make more sense when we are in a system test, because other test could expect that this text is part of the output.